token_transfer_transaction_unit_test.go

```diff
@@ -8,6 +8,50 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestUnitTokenTransferTransactionTransfers(t *testing.T) {
+	amount := NewHbar(1)
+	accountID1 := AccountID{Account: 3}
+	accountID2 := AccountID{Account: 4}
+	tokenID1 := TokenID{Token: 5}
+	tokenID2 := TokenID{Token: 6}
+	tokenID3 := TokenID{Token: 7}
+	tokenID4 := TokenID{Token: 8}
+	nftID1 := NftID{TokenID: tokenID3, SerialNumber: 9}
+	nftID2 := NftID{TokenID: tokenID4, SerialNumber: 10}
+
+	tokenTransfer := NewTransferTransaction().
+		AddHbarTransfer(accountID1, amount).
+		AddHbarTransfer(accountID2, amount.Negated()).
+		AddTokenTransfer(tokenID1, accountID1, 10).
+		AddTokenTransfer(tokenID1, accountID2, -10).
+		AddTokenTransfer(tokenID2, accountID1, 10).
+		AddTokenTransfer(tokenID2, accountID2, -10).
+		AddNftTransfer(nftID1, accountID1, accountID2).
+		AddNftTransfer(nftID2, accountID2, accountID1)
+
+	hbarTransfers := tokenTransfer.GetHbarTransfers()
+	tokenTransfers := tokenTransfer.GetTokenTransfers()
+	nftTransfers := tokenTransfer.GetNftTransfers()
+
+	assert.Equal(t, len(hbarTransfers), 2)
+	assert.Equal(t, hbarTransfers[accountID1], amount)
+	assert.Equal(t, hbarTransfers[accountID2], amount.Negated())
+
+	assert.Equal(t, len(tokenTransfers), 2)
+	assert.Equal(t, len(tokenTransfers[tokenID1]), 2)
+	assert.Equal(t, tokenTransfers[tokenID1][0], TokenTransfer{AccountID: accountID1, Amount: 10})
+	assert.Equal(t, tokenTransfers[tokenID1][1], TokenTransfer{AccountID: accountID2, Amount: -10})
+	assert.Equal(t, len(tokenTransfers[tokenID2]), 2)
+	assert.Equal(t, tokenTransfers[tokenID2][0], TokenTransfer{AccountID: accountID1, Amount: 10})
+	assert.Equal(t, tokenTransfers[tokenID2][1], TokenTransfer{AccountID: accountID2, Amount: -10})
+
+	assert.Equal(t, len(nftTransfers), 2)
+	assert.Equal(t, len(nftTransfers[tokenID3]), 1)
+	assert.Equal(t, nftTransfers[tokenID3][0], TokenNftTransfer{SenderAccountID: accountID1, ReceiverAccountID: accountID2, SerialNumber: 9})
+	assert.Equal(t, len(nftTransfers[tokenID4]), 1)
+	assert.Equal(t, nftTransfers[tokenID4][0], TokenNftTransfer{SenderAccountID: accountID2, ReceiverAccountID: accountID1, SerialNumber: 10})
```

Comment:
nit, can make the asserts more organized

```suggestion
	assert.Equal(t, hbarTransfers, map[AccountID]Hbar{
		accountID1: amount,
		accountID2: amount.Negated(),
	})

	assert.Equal(t, tokenTransfers, map[TokenID][]TokenTransfer{
		tokenID1: {
			{AccountID: accountID1, Amount: 10},
			{AccountID: accountID2, Amount: -10},
		},
		tokenID2: {
			{AccountID: accountID1, Amount: 10},
			{AccountID: accountID2, Amount: -10},
		},
	})

	assert.Equal(t, nftTransfers, map[TokenID][]TokenNftTransfer{
		tokenID3: {{SenderAccountID: accountID1, ReceiverAccountID: accountID2, SerialNumber: 9}},
		tokenID4: {{SenderAccountID: accountID2, ReceiverAccountID: accountID1, SerialNumber: 10}},
	})
```